test/unit/search/views/minimiser_test.cpp

```diff
@@ -242,3 +242,42 @@ TEST_F(minimiser_test, two_ranges_unequal_size)
 {
     EXPECT_THROW((seqan3::detail::minimiser_view{text1 | kmer_view, text3 | rev_kmer_view, 5}), std::invalid_argument);
 }
+
+TEST_F(minimiser_test, base)
+{
+    std::vector<size_t> hashes{3, 6, 5, 4, 8, 4, 4, 2, 5, 4};
+    using hash_iterator_t = typename std::vector<size_t>::iterator;
+
+    hash_iterator_t const hash_begin = hashes.begin();
+    hash_iterator_t const hash_end = hashes.end();
+
+    size_t const window_size = 5;
+    auto const minimiser = seqan3::views::minimiser(hashes, window_size);
+    auto minimiser_it = minimiser.begin();
+
+    // hash iterator is at position window_size - 1 and
+    // points to the last element in the window
+    // index:   0, 1, 2, 3, 4,  5, 6, 7, 8, 9
+    // hashes: [3, 6, 5, 4, 8], 4, 4, 2, 5, 4
+    //                      ^
+    EXPECT_EQ(minimiser_it.base() - hash_begin, window_size - 1);
+
+    // points to the last element in the window
+    // index:  0,  1, 2, 3, 4, 5,  6, 7, 8, 9
+    // hashes: 3, [6, 5, 4, 8, 4], 4, 2, 5, 4
+    //                         ^
+    ++minimiser_it;
+    EXPECT_EQ(minimiser_it.base() - hash_begin, 5u);
+
+    // points to the last element in the window
+    // index:  0, 1, 2,  3, 4, 5, 6, 7,  8, 9
+    // hashes: 3, 6, 5, [4, 8, 4, 4, 2], 5, 4
+    //                               ^
+    ++minimiser_it;
+    EXPECT_EQ(minimiser_it.base() - hash_begin, 7u);
```

Comment:
I don`t understand how you jump over the window starting at index 2, I would have expected another ++minimiser_it; but the test is running through, what am I missing? :sweat_smile:

Reply:
Not sure, why you think that, but the minimiser view has a new minimiser only if the current minimal element in the window "expires" (either by a new overall minimal value or if the current minimal element leaves the window).

```
// index:  0 1 2 3 4 5 6 7 8 9
// hashes:[3 6 5 4 8]4 4 2 5 4 (new minimiser: starting window)
// new min:*
// hashes: 3[6 5 4 8 4]4 2 5 4 (new minimiser: current min is outside of window, recompute min)
// new min:          *
// hashes: 3 6[5 4 8 4 4]2 5 4 (no change, no reported minimiser)
// hashes: 3 6 5[4 8 4 4 2]5 4 (new minimiser: new value is smaller than old min)
// new min:              *
// hashes: 3 6 5 4[8 4 4 2 5]4 (no change, no reported minimiser)
// hashes: 3 6 5 4 8[4 4 2 5 4](no change, no reported minimiser)
```

Reply:
Ah, I didn`t realize the minimiser changed. Sorry. Thanks a lot!